src/gas_saved_2.py

```diff
@@ -0,0 +1,63 @@
+import os
+
+import click
+import pandas as pd
+from dotenv import load_dotenv
+from web3 import Web3
+
+from src.db.pg_client import pg_engine
+
+from duneapi.api import DuneAPI
+from duneapi.types import DuneQuery, Network
+from duneapi.util import open_query
+
+
+def fetch_trades_per_batch(dune: DuneAPI):
+    query = DuneQuery.from_environment(
+        raw_sql=open_query("./queries/num-trades-per-batch.sql"),
+        name="num-trades-per-batch",
+        network=Network.MAINNET,
+        parameters=[],
```

Comment:
FYI - these fields are all Optional now with defaults essentially as you have specified here (empty parameters and mainnet and empty string for name). No need to change here, just thought you might like to know about these recently added conveniences.